### PR TITLE
Feat migrate to tsdoc

### PR DESCRIPTION
**Pull request by jsolly**
Converted all jsdoc to tsdoc. Added the tsdoc plugin for eslint.
https://tsdoc.org/

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -4,17 +4,20 @@ module.exports = {
...
   },
   'plugins': [
-    '@typescript-eslint',
+    '@typescript-eslint/eslint-plugin',
+    'eslint-plugin-tsdoc',
   ],
   'rules': {
     'max-len': 'off',
+    'tsdoc/syntax': 'warn',
+    '@typescript-eslint/no-var-requires': 0,
```
Comment (bouvens) [inline .eslintrc.js:21]: I also add
`'@typescript-eslint/explicit-function-return-type': 'warn',`
but it's a matter of taste

**Review by bouvens — APPROVED** [review]

File: .prettierrc [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+  "semi": true,
+  "trailingComma": "none",
```
Comment (bouvens) [inline .prettierrc:3]: It's not a problem, but trailingComma makes diffs cleaner

File: src/asteroids.ts [hunk 3]
```diff
@@ -71,13 +71,13 @@ function newAsteroid(x:number, y:number, r:number) {
  * @returns Array of Asteroids
  */
 function getRoidsInfo() {
-  return {roids: roids, roidsLeft: roidsLeft, roidsTotal: roidsTotal};
+  return { roids: roids, roidsLeft: roidsLeft, roidsTotal: roidsTotal };
```
Comment (bouvens) [inline src/asteroids.ts:74]: We can simply write
```suggestion
  return { roids, roidsLeft, roidsTotal };
```
That's how modern JS works. We can also add an ESLint rule for that. 😉

File: src/ship.ts [hunk 4]
```diff
@@ -40,7 +40,7 @@ class Ship {
    * @param lives - Create a ship with a given number of lives
    * @param blinkOn - Determine if ship should be blinking or not
    */
-  constructor(lives:number=START_LIVES, blinkOn = false) {
+  constructor(lives: number = START_LIVES, blinkOn = false) {
```
Comment (bouvens) [inline src/ship.ts]: That's so beautiful 🥲

**Review by bouvens — APPROVED** [review]

File: .eslintrc.js (hunk 1 above)
Comment (jsolly) [reply .eslintrc.js]: Done! I added this and return types for all the functions.

File: .prettierrc (hunk 2 above)
Comment (jsolly) [reply .prettierrc]: I agree! Fixed in this commit
https://github.com/jsolly/GeoAsteroids/commit/828840a103078d50a3d1c16c5a78a4b2c757f7fc

File: src/asteroids.ts (hunk 3 above)
Comment (jsolly) [reply src/asteroids.ts]: Nice. I added the return type and it looks really clean!
```ts
function getRoidsInfo(): {
  roids: typeof roid[];
  roidsLeft: number;
  roidsTotal: number;
} {
  return { roids, roidsLeft, roidsTotal };
}
```

**Comment by jsolly** [comment]
Thank you @bouvens for all your comments. I have implemented all of them 🚀

We are using prettier now and all doc strings are following the tsdoc spec. There's still work to do (I still see some empty doc strings), but will handle that in new PRs/commits.

**Comment by bouvens** [comment]
> Thank you @bouvens for all your comments. I have implemented all of them 🚀
> 
> We are using prettier now and all doc strings are following the tsdoc spec. There's still work to do (I still see some empty doc strings), but will handle that in new PRs/commits.

You are always welcome 😄

**Pull request merged**